### PR TITLE
Add UIBarButtonItem extensions

### PR DESCRIPTION
**Pull request by asmallteapot**
#### Checklist
- [x] Updated CHANGELOG.md.

### CODE REVIEW

**Comment by mdiep** [comment]
There are some compiler errors in the tests. 🙈

**Comment by asmallteapot** [comment]
@mdiep: Ta! I've rebased this off `master` and it looks like everything is green now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 1. Add `becomeFirstResponder` and `resignFirstResponder` extensions to `UIResponder`. (#3585, kudos to @Marcocanc)
 2. Added `title` binding target to `UIViewController` (#3588, kudos to @cocoahero).
 3. Added several trigger signals for view lifecycle events to `UIViewController` (#3588, kudos to @cocoahero).
+4. Add extensions for several properties on `UIBarButtonItem` (#3586, kudos to @asmallteapot).
```
Comment (mdiep) [inline CHANGELOG.md]: This needs to go _above_ `8.0.0-rc.1` since that's already be released. 🙈

**Comment by asmallteapot** [comment]
Womp womp. Fixed now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # master
 *Please put new entries at the top.
 
+# 8.0.0-rc.2
```
Comment (mdiep) [inline CHANGELOG.md]: Sorry, still not quite right. 😞 This shouldn't have a header. We add the header when we make the release. The next release may or not be rc2; we don't want to assume.

**Comment by mdiep** [comment]
Thanks for the PR!

**Pull request merged**